### PR TITLE
sensor: Add Silabs SI7055 support

### PR DESCRIPTION
**Pull request by greg-leach**
Add support for the Silabs SI7055 temperature sensor

Signed-off-by: Greg Leach <greg.leach@lairdconnect.com>

### CODE REVIEW

**Review by rerickson1 — APPROVED** [review]

File: drivers/sensor/si7055/si7055.c [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+{
+	int retval;
+	#if CONFIG_SI7055_ENABLE_CHECKSUM
+	uint8_t temp[SI7055_TEMPERATURE_READ_WITH_CHECKSUM_SIZE];
+	#else
+	uint8_t temp[SI7055_TEMPERATURE_READ_NO_CHECKSUM_SIZE];
+	#endif
+
+	retval = i2c_burst_read(i2c_dev, DT_INST_REG_ADDR(0),
+		SI7055_MEAS_TEMP_MASTER_MODE, temp, sizeof(temp));
```
Comment (avisconti) [inline drivers/sensor/si7055/si7055.c:44]: check identation

File: drivers/sensor/si7055/si7055.c [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+	/* Refer to
+	 * https://www.silabs.com/documents/public/data-sheets/Si7050-1-3-4-5-A20.pdf
+	 */
+
+	if (retval == 0) {
+		#if CONFIG_SI7055_ENABLE_CHECKSUM
+		if (crc8(temp, SI7055_DATA_SIZE, SI7055_CRC_POLY,
+				SI7055_CRC_INIT, false) !=
+				temp[SI7055_DATA_SIZE]){
```
Comment (avisconti) [inline drivers/sensor/si7055/si7055.c:54]: identation

File: drivers/sensor/si7055/si7055.c [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+					LOG_ERR("checksum failed.\n");
+					return(-EIO);
+		}
+		#endif
+		si_data->temperature = (temp[SI7055_TEMPERATURE_DATA_BYTE_0]
+					<< 8) |
+					temp[SI7055_TEMPERATURE_DATA_BYTE_1];
```
Comment (avisconti) [inline drivers/sensor/si7055/si7055.c:61]: identation (I would put the <<8 together with first line)

File: drivers/sensor/si7055/si7055.c [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+	 * https://www.silabs.com/documents/public/data-sheets/Si7050-1-3-4-5-A20.pdf
+	 */
+
+	if (chan == SENSOR_CHAN_AMBIENT_TEMP) {
+
+		int32_t temp_ucelcius =
+			(((SI7055_CONV_FACTOR_1 *
+				(int32_t)si_data->temperature)
+			/ (__UINT16_MAX__ + 1)) - SI7055_CONV_FACTOR_2) *
+				SI7055_MULTIPLIER;
```
Comment (avisconti) [inline drivers/sensor/si7055/si7055.c:104]: identation also here

**Review by avisconti — DISMISSED** [review]
Seems overall simple and good, but check identation when you break lines

File: drivers/sensor/si7055/Kconfig [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+	bool "Si7055 Temperature Sensor"
+	depends on (I2C && HAS_DTS_I2C)
+	help
+	  Enable I2C-based driver for Si7055 Temperature Sensor.
+
+if SI7055
+
+config SI7055_ENABLE_CHECKSUM
+	bool "SI7055 Checksum On Measurement"
+	default n
```
Comment (MaureenHelm) [inline drivers/sensor/si7055/Kconfig:15]: Bools are implicitly default n, so you can remove this line

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please add this sensor to `tests/drivers/build_all/sensors_i_z.conf` to ensure it gets build in CI.

I don't quite understand your comment about squashed commits in the commit message. You seem to refer to a commit that isn't in the history?

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**